JavaScript/chapter02/p06_palindrome/avc278.js

```diff
@@ -0,0 +1,88 @@
+// Palindrome: Implement a function to check if a linked list is a palindrome.
+
+const assert = require("assert");
+const { arrayToLinkedList } = require("../../lib/avc278/linkedlist");
+
+/**
+ * Checks whether the input linked list is a palindrome
+ * @param   {LinkedListNode} list input linked list to check against
+ * @return  {boolean}             whether the input linked list is a palindrome
+ *
+ * For this problem, we start by finding the midpoint idx of the linked list. This traversal takes O(N) runtime and O(1)
+ * space as we only store a counter, and a pointer to the location within the linked list. We then perform another
+ * traversal to point to the midpoint, and reverse the linked list from that point onwards. From this point onwards, we
+ * only need to compare the two linked lists from the midpoint to the end, and from the start to the midpoint.
+ * All in all, since we only perform traversals without any inner traversals, the runtime at any traversal is O(N),
+ * where N is the length of the linked list, and O(1) space as we store pointers.
+ * Runtime: O(N)
+ * Space:   O(1)
+ *
+ */
+const reverse = (node) => {
+  let prev;
+  while (node !== null) {
+    const next = node.next;
+    node.next = prev;
+    prev = node;
+    node = next;
+  }
+  return prev;
+};
+
+const isPalindrome = (list) => {
+  let fast = list;
+  let slow = list;
+  let counter = 0;
+  while (fast !== null) {
+    fast = fast.next;
+    counter += 1;
+  }
+  counter /= 2;
+
+  while (counter > 0) {
+    slow = slow.next;
+    counter -= 1;
+  }
```

Comment:
maybe

```suggestion
  for (let i = 0; i < counter / 2; i++) {
    slow = slow.next;
  }
```

since the number of times this loop needs to run is known beforehand.